rust/lon/src/sources.rs

```diff
@@ -122,6 +148,70 @@ impl Source {
             Self::GitHub(s) => s.frozen,
         }
     }
+
+    pub fn diff(&self, summary: &UpdateSummary, num_commits: usize) -> Result<String> {
+        match self {
+            Self::Git(s) => git::diff_history(
+                &s.url,
+                summary.old_revision.as_str(),
+                summary.new_revision.as_str(),
+                num_commits,
+            ),
+            Self::GitHub(s) => {
```

Comment:
What's the benefit of calling the API here? I'd generally want to avoid calling the API and using generic git functions as much as possible.

Reply:
It is much faster for nixpkgs e.g. if you need to fetch the git data it would take a lot more data and time (as it is in CI it might not be that bad ig ?)